PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,349 @@
+    }
+    // selection on D0Candidate
+    if (!selectionTopolD0(d0Candidate)) {
+      return false;
+    }
+    return true;
+  }
+
+  /// @brief Conjugate-dependent topological cuts
+  /// @tparam T1 Table iterator type of D0 Candidate
+  /// @tparam T2 Table iterator type of Dstar Candidate
+  /// @param d0candidate D0 candidate object
+  /// @param dstarCandidate Dstar candidate object
+  /// @return true/false depending on success of selection
+  template <typename T1, typename T2>
+  bool selectionTopolConjugate(const T1& d0candidate, const T2& dstarCandidate)
+  {
+    auto dstarpT = dstarCandidate.pt();
+    auto pTBin = findBin(binsPtForDstar, dstarpT);
+    if (pTBin == -1) {
+      return false;
+    }
+    auto SoftPi = dstarCandidate.template prongPi_as<TracksSel>();
+
+    if (SoftPi.sign() > 0.) { // Selection of D*+
+      // auto Mpipik = std::array{massPi,massPi,massK};
+      // auto InvDstar = dstarCandidate.dstarInvMass(Mpipik);
+      auto InvDstar = hfHelper.invMassDstar(dstarCandidate);
+
+      // auto Mpik = std::array{massPi,massK};
+      // auto InvD0 = d0candidate.d0m(Mpik);
+      auto InvD0 = hfHelper.invMassD0ToPiK(d0candidate);
+
+      // LOGF(info,"+ %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > cutsForD0->get(pTBin, "d0pi") || std::abs(d0candidate.impactParameter1()) > cutsForD0->get(pTBin, "d0K")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(hfHelper.cosThetaStarD0(d0candidate)) > cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+
+    } else if (SoftPi.sign() < 0) { // Selection of D*-
+      // auto Mpikpi = std::array{massPi,massK,massPi};
+      // auto InvDstar = dstarCandidate.dstarInvMass(Mpikpi);
+      auto InvDstar = hfHelper.invMassAntiDstar(dstarCandidate);
+
+      // auto Mkpi = std::array{massK,massPi};
+      // auto InvD0 = d0candidate.d0m(Mkpi);
+      auto InvD0 = hfHelper.invMassD0barToKPi(d0candidate);
+
+      // LOGF(info,"- %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > cutsForD0->get(pTBin, "d0K") || std::abs(d0candidate.impactParameter1()) > cutsForD0->get(pTBin, "d0pi")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(hfHelper.cosThetaStarD0bar(d0candidate)) > cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void process(TracksSel const&, aod::Hf2Prongs const&, aod::HfD0fromDstar const& rowsD0cand, aod::HfDstarCand const& rowsDstarCand)
```

Comment:
If I am not mistaken we do not need to consume the `aod::Hf2Prongs` table
```suggestion
  void process(TracksSel const&, aod::HfD0fromDstar const& rowsD0cand, aod::HfDstarCand const& rowsDstarCand)
```

Reply:
```aod::HfDstarCand```  table has a column of ```ProngD0```, which points to ```aod::Hf2Prong```  Table. And as much I remember you earlier told that sometimes this table can have collision information different from that in ```HfDstar```, so you suggested in a meeting to keep it. If we want to remove it from the subscription, we will also have to remove ```ProngD0``` column from ```HfDstarCand``` table.